test/notificationSpec.js

```diff
@@ -321,305 +295,41 @@ describe('Notifications', function () {
         .end()
     })
 
-    it('should submit the source transfer corresponding to the destination transfer it is notified about if the execution conditions are the same', function * () {
-      const payment = this.formatId(this.paymentSameExecutionCondition,
-        '/payments/')
-
-      nock(payment.destination_transfers[0].id)
-        .put('')
-        .reply(201, _.assign({}, payment.destination_transfers[0], {
-          state: 'prepared'
-        }))
-
-      let sourceTransferExecuted = nock(payment.source_transfers[0].id)
-        .put('/fulfillment',
-          this.notificationWithConditionFulfillment
-            .resource.execution_condition_fulfillment)
-        .reply(201, _.assign({}, payment.source_transfers[0], {
-          state: 'executed'
-        }))
+    ;[
+      {
+        label: 'should return 200 if the payment is not relevant to the connector',
+        error: 'UnrelatedNotificationError'
+      }, {
+        label: 'should return 200 if the rate of the payment is worse than the one currently offered',
+        error: 'UnacceptableRateError'
+      }, {
+        label: 'should return 200 if the payment is from an unknown source ledger',
+        error: 'AssetsNotTradedError'
+      }, {
+        label: 'should return a 200 if the source transfer is expired',
+        error: 'UnacceptableExpiryError'
+      }
+    ].forEach(function (data) {
+      it(data.label, function * () {
+        this.ledgers.getLedger(this.notificationSourceTransferPrepared.resource.ledger)
```

Comment:
I think it's more readable to spell each of these out rather than getting fancy...